compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -486,6 +532,133 @@ abstract class GenJSCode extends plugins.PluginComponent
       classDefinition
     }
 
+    /** Generate an instance of an anonymous Scala.js defined class inline
+     *
+     *  @param sym Class to generate the instance of
+     *  @param args Arguments to the constructor
+     *  @param pos Position of the original New tree
+     */
+    def genAnonSJSDefinedNew(sym: Symbol, args: List[js.Tree],
+        pos: Position): js.Tree = {
+      assert(isScalaJSDefinedAnonJSClass(sym),
+          "Generating AnonSJSDefinedNew of non anonymous SJSDefined JS class")
+
+      // Find the ClassDef for this anonymous class
+      val classDef = consumeLazilyGeneratedAnonClass(sym)
+
+      // Generate a normal SJSDefinedJSClass
+      val origJsClass =
+        nestedGenerateClass(sym)(genScalaJSDefinedJSClass(classDef))
+
+      // Partition members into the ones we inline and the ones we keep
+      val retainedMembers = ListBuffer.empty[js.Tree]
+      var constructor: Option[js.MethodDef] = None
+      val exportedMembers = ListBuffer.empty[js.Tree]
+
+      origJsClass.defs.foreach {
+        case _: js.FieldDef => // ignore. we get it from the ctor
```

Comment:
You actually need to keep them, because of fields with no initializer (`var f: Int = _`)